tasks/hacbs-java.yaml

```diff
@@ -0,0 +1,269 @@
+        <proxies>
+          <!-- The proxies added here are generated from environment variables. Don't change. -->
+          <!-- ### HTTP proxy from ENV ### -->
+        </proxies>
+      </settings>
+      EOF
+
+      xml=""
+      if [ -n "$(params.PROXY_HOST)" -a -n "$(params.PROXY_PORT)" ]; then
+        xml="<proxy>\
+          <id>genproxy</id>\
+          <active>true</active>\
+          <protocol>$(params.PROXY_PROTOCOL)</protocol>\
+          <host>$(params.PROXY_HOST)</host>\
+          <port>$(params.PROXY_PORT)</port>"
+        if [ -n "$(params.PROXY_USER)" -a -n "$(params.PROXY_PASSWORD)" ]; then
+          xml="$xml\
+              <username>$(params.PROXY_USER)</username>\
+              <password>$(params.PROXY_PASSWORD)</password>"
+        fi
+        if [ -n "$(params.PROXY_NON_PROXY_HOSTS)" ]; then
+          xml="$xml\
+              <nonProxyHosts>$(params.PROXY_NON_PROXY_HOSTS)</nonProxyHosts>"
+        fi
+        xml="$xml\
+            </proxy>"
+        sed -i "s|<!-- ### HTTP proxy from ENV ### -->|$xml|" settings.xml
+      fi
+
+      if [ -n "$(params.SERVER_USER)" -a -n "$(params.SERVER_PASSWORD)" ]; then
+        xml="<server>\
+          <id>serverid</id>"
+        xml="$xml\
+              <username>$(params.SERVER_USER)</username>\
+              <password>$(params.SERVER_PASSWORD)</password>"
+        xml="$xml\
+            </server>"
+        sed -i "s|<!-- ### SERVER's USER INFO from ENV ### -->|$xml|" settings.xml
+      fi
+
+      if [ -n "$(params.MAVEN_MIRROR_URL)" ]; then
+        xml="    <mirror>\
+          <id>mirror.default</id>\
+          <url>$(params.MAVEN_MIRROR_URL)</url>\
+          <mirrorOf>*</mirrorOf>\
+        </mirror>"
+        sed -i "s|<!-- ### mirrors from ENV ### -->|$xml|" settings.xml
+      fi
+
+  - name: mvn-goals
+    image: $(params.MAVEN_IMAGE)
+    workingDir: $(workspaces.source.path)/$(params.CONTEXT_DIR)
+    command: ["/usr/bin/mvn"]
+    args:
+      - -s
+      - settings.xml
+      - "$(params.GOALS)"
+      - -Dmaven.test.skip
+  - name: analyse-dependencies
+    image: $(params.DEPENDENCY_ANALYSER_IMAGE)
+    args:
+      - path
+      - $(workspaces.source.path)/$(params.CONTEXT_DIR)/$(params.DEPENDENCY_ANALYSER_PATH)
+  - name: docker-settings
+    image: registry.access.redhat.com/ubi8/ubi-minimal:8.5
+    workingDir: $(workspaces.source.path)/$(params.CONTEXT_DIR)
+    script: |
+      #!/usr/bin/env bash
+
+      cat > Dockerfile.gen <<EOF
+      FROM registry.access.redhat.com/ubi8/openjdk-17:1.12
+
+      ENV LANG='en_US.UTF-8' LANGUAGE='en_US:en'
+
+
+      # We make four distinct layers so if there are application changes the library layers can be re-used
+      COPY --chown=185 target/* /deployments/
+
+      EXPOSE 8080
+      USER 185
+      ENV AB_JOLOKIA_OFF=""
+      ENV JAVA_OPTS="-Dquarkus.http.host=0.0.0.0 -Djava.util.logging.manager=org.jboss.logmanager.LogManager"
+      ENV JAVA_APP_JAR="/deployments/quarkus-run.jar"
```

Comment:
Yop, exactly, now working with https://github.com/devfile-samples/devfile-sample-code-with-quarkus and it should be improved to support everything somehow.

Reply:
Looking at this I think this is broken, we are copying the whole /target directory to `/deployments`, but then running  `/deployments/quarkus-run.jar`. For this simple case we should be copying `target/quarkus-app`.

As I said above though this is really limiting, it will only work with single module Quarkus applications that use the default packaging. 

We are probably going to need to add some configuration options here to control what gets copied and what gets run to allow for other application types, at least as an initial option. The path that gets copied is also what should be passed into the dependency analyser path above.